taipy/core/cycle/cycle.py

```
@@ -157,7 +157,7 @@ def __getattr__(self, attribute_name):
         raise AttributeError(f"{attribute_name} is not an attribute of cycle {self.id}")
 
     def __eq__(self, other):
-        return self.id == other.id
+        return isinstance(other, Cycle) and self.id == other.id
```

should we raise an error instead of just False if the `other` is not a Cycle?

---

I would not raise an error. This code looks correct to me.

---

same, I wouldn't raise an error also, the purpose is to check if this `other` value is equal to the Cycle, if it's not, then the right thing to do is return False. We don't need to raise any errors for it